app/controllers/countries_controller.rb

```diff
@@ -64,7 +64,8 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_country
-      @country = Country.find(params[:id])
+      #see https://hackhands.com/ruby-rails-performance-tuning/
+      @country = Country.includes( :percentages => :languages).find(params[:id])
```

Comment:
We should try to use the Ruby 1.9+ syntax, assuming it doesn't break this call:
`@country = Country.includes(percentages: :languages).find(params[:id])`

So you're loading the percentage and languages for the country here, where are you then using the associated objects you just retrieved?